### ISSUE TITLE
Map view rotation feature

### ISSUE THREAD

**Issue by srbhagwatsrb** [peterLaurence/MapView#2]
I am building a POC for Indoor navigation and i came across TIleview as well as MapView. I would like to use Kotlin version. But i did not find demo on having a rotatable map. Is it possible to rotate the map? 

What i want is that in my indoor navigation app, based on magnetic field directions, map should get adjusted/rotated in order to show FPP view. (First person perspective view). Do you have any info on how this can be achieved?

**Comment by p-lr** [comment]
I would use the `setRotation` method available on every `View` since api level 11. Apply to the `MapView` or one of it's parent view in your app.
Link to doc: https://developer.android.com/reference/android/view/View.html#setRotation%28float%29

**Comment by srbhagwatsrb** [comment]
Hmm but I need something like Google map rotation. On drag and rotate on a zoompanlayout. Could you please guide?

**Comment by p-lr** [comment]
In the meantime, I had more time to think about it.
The proper way to support this would require some adaptations of the library. Indeed, the rotated map needs to update the visible tiles, which cannot be achieved by simply rotating the MapView.
So MapView currently doesn't support this feature. Although it might be added the future, either when I have time to do this or someone makes a PR.

**Comment by srbhagwatsrb** [comment]
RIght i was thinking about the same. It's really a good feature to add. And may be i will have to use an old version of tileview (2.x) where someone have done this. I do not want to use an old version again. Any chance this feature can be implemented soon in this also?

**Comment by p-lr** [comment]
I can't make promises. But when it's done, it will be very easy for you to
migrate, as TileView 2 API is very similar to MapView. Basically you have a
BitmapProvider instead of TileStreamProvider, and have to define manually
the levels.

Note to self: a [GestureDetector](https://github.com/mapbox/mapbox-gl-native/blob/1b582147534cccf6b81bcdc66774942e9ce73cbe/platform/android/MapboxGLAndroidSDK/src/main/java/com/almeros/android/multitouch/gesturedetectors/TwoFingerGestureDetector.java)

**Comment by srbhagwatsrb** [comment]
Thanks @peterLaurence Do try to add this soon as it will be extremely good to use. This map view is already in production grade. So with rotatable map it will work great in all real time apps. Great work on this library.

**Comment by p-lr** [comment]
I have a very promising proof of concept of rotating map in my `rotation` branch. But it's not ready yet.
Will you be available to beta test when it's done? (I hope soon)

**Comment by shriharsha-bhagwat** [comment]
Sure i will be ready to test it out. Let me know.

**Comment by p-lr** [comment]
Perfect. I'm making progress - paths rotation and a few optimizations need to be done.
I let you know. Thanks

**Comment by p-lr** [comment]
@android-bash 
It has been harder and longer than I thought. 
I've published the version `2.0.0-beta04`

The API around map rotation is stable (although I might make slight changes before final release).
2.x.x is a major shift from 1.x.x as it enables map rotation. It's only an opt-in feature, while configuring the MapView, using enableRotation(). The RotatingMapFragment from the demo is an example. 

Beware that there are some breaking changes. Most of the time, it's just a package name change.
The ScaleChangeListener interface has been removed. If you relied on this, I can explain how to do the equivalent in 2.x.x
